### DISCUSSION TITLE
duckdns work with immich?

### DISCUSSION THREAD

**Discussion by hollow-frenk** [immich-app/immich#1551] [Q&A]
Hello, maybe they have already asked but I have not found anything, how do I access the server from outside the house? duckdns?

**Comment by bo0tzz** [comment]
If you search for "reverse proxy" in the discussions here you'll find a few threads on the topic: https://github.com/immich-app/immich/discussions?discussions_q=reverse+proxy

**Reply by hollow-frenk** [reply]
Thank you, but in practice, how do I do it? I’m not very experienced...

**Reply by bo0tzz** [reply]
I recommend you have a search for some guides on how to set up a reverse proxy, as it'll depend on the details of your setup etc.

**Comment by jpmiller25** [comment]
There is some decent complication to setting up a reverse proxy, which could potentially open security vulnerabilities. I'd caution you to research your solution carefully.
One solution would be to set up a local VPN server, then your clients can basically securely connect to your local network wherever they are. WireGuard is one solution, check out https://hub.docker.com/r/weejewel/wg-easy, or look into your router, the better ones may have a good VPN solution integrated. 
Another solution would be to use a ddns like duckdns to point to your homes external ip address, or buy a domain that you can point to your address. You'd have to port forward 80 and 443 through your firewall to your server and run a reverse proxy that takes care of certificates, ssl encryption, etc. Look at caddy, traefik, SWAG, nginx.
One other option, which I actually do, is I have a domain with cloudflare and I use something called an Argo tunnel. I run https://hub.docker.com/r/cloudflare/cloudflared in a container which connects to cloudflare, and they handle all proxy, encryption, etc. Pros are much less configuration on my end, and my homes external ip is never exposed, and no port forwarding whatsoever. Cons are I'm putting a decent amount of trust in cloudflare, also there is an issue where large files (long videos) don't get uploaded unless I'm local, they have some sort of restriction that I can't change.
Good luck and be safe!
